data/data.yaml

```
@@ -1270,3 +1270,162 @@
      • Регулярное выражение должно оставаться понятным и легким для поддержки. Сложные, трудно читаемые выражения могут затруднить их использование и усложнить внесение изменений, повышая риск ошибок.
      • При использовании механизма недетерминированного конечного автомата (НКА) выражение должно быть эффективно и приводить к быстрому результату. Однако стоит учитывать, что конструкции с итерацией Клини (* и +) особенно в начале выражения могут усилить недетерминизм, что повышает вероятность долгого перебора вариантов и ухудшает производительность. Хорошо составленное регулярное выражение должно минимизировать такие конструкции, если это возможно, чтобы избежать избыточной сложности и не затягивать обработку.
   author: Еделькин Герман
+
+- question: >
+    Определение производной языка по слову. 
+    Производные Бржозовски (Brzozowski derivative)
+  answer: > 
+    Производная Бржозовски (Brzozowski derivative) языка L по слову u - это множество u^(-1)L = {w|uw ∈ L} слов, 
+    которые получены из слов языка L, путём удаления в них префикса равного слову u. 
+    Верно следующее: ε ∈ u^(-1)L => u ∈ L, то есть, если пустое слово принадлежит производной Бржозовски языка L по слову u, 
+    то из это следует, что слово u принадлежит языку L. На практике производные Бржозовски (Brzozowski derivative) можно 
+    использовать для проверки того, принимает ли регулярное выражение произвольное слово или нет.
+  author: Утебаева Милена
+
+- question: >
+    Определение степени неоднозначности недетерминированного конечного автомата (Nondeterministic Finite Automaton). 
+    Виды неоднозначностей: полиномиальная неоднозначность, экспоненциальная неоднозначность
+  answer: > 
+    Степень недетерминизма НКА (NFA), обозначим автомат буквой A, - это функция f(n), зависящая от натурального 
+    числа n, такая, что её значение на аргументе n равно наибольшему возможному числу путей разбора слова длины n в автомате A.
+    Если для любого натурального числа n значение функции f(n) равное единице, то автомат A называется однозначным.
+    Если существует некоторое натуральное число N, такое, что для любого натурального числа n значение функции f(n)
+    не превосходит числа N, то автомат A называется конечно неоднозначным.
+    Если существует натуральное k, такое, что значение функция f(n) оценивается как O(n^k), 
+    то автомат A называется полиномиально неоднозначным.
+    Если существует натуральное k, такое, что значение функция f(n) оценивается как O(k^n), 
```

```suggestion
    Если существует натуральное k, такое, что значение функции f(n) оценивается как O(k^n), 
```

и тут вроде тоже вот так